File: src/main/java/dev/robothanzo/werewolf/config/WebSocketConfig.java
package dev.robothanzo.werewolf.config;

import dev.robothanzo.werewolf.security.GlobalWebSocketHandler;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.socket.config.annotation.EnableWebSocket;
import org.springframework.web.socket.config.annotation.WebSocketConfigurer;
import org.springframework.web.socket.config.annotation.WebSocketHandlerRegistry;

@Configuration
@EnableWebSocket
public class WebSocketConfig implements WebSocketConfigurer {

    private final GlobalWebSocketHandler globalWebSocketHandler;

    public WebSocketConfig(GlobalWebSocketHandler globalWebSocketHandler) {
        this.globalWebSocketHandler = globalWebSocketHandler;
    }

    @Override
    public void registerWebSocketHandlers(WebSocketHandlerRegistry registry) {
        registry.addHandler(globalWebSocketHandler, "/ws")
                .addInterceptors(new org.springframework.web.socket.server.support.HttpSessionHandshakeInterceptor())
                .setAllowedOrigins(
                        "http://localhost:5173",
                        "https://wolf.robothanzo.dev",
                        "http://wolf.robothanzo.dev");
    }
}

File: src/main/kotlin/dev/robothanzo/werewolf/config/WebSocketConfig.kt
package dev.robothanzo.werewolf.config

import dev.robothanzo.werewolf.security.GlobalWebSocketHandler
import org.springframework.context.annotation.Configuration
import org.springframework.web.socket.config.annotation.EnableWebSocket
import org.springframework.web.socket.config.annotation.WebSocketConfigurer
import org.springframework.web.socket.config.annotation.WebSocketHandlerRegistry
import org.springframework.web.socket.server.support.HttpSessionHandshakeInterceptor

@Configuration
@EnableWebSocket
class WebSocketConfig(private val globalWebSocketHandler: GlobalWebSocketHandler) :
    WebSocketConfigurer {

    override fun registerWebSocketHandlers(registry: WebSocketHandlerRegistry) {
        registry.addHandler(globalWebSocketHandler, "/ws")
            .addInterceptors(HttpSessionHandshakeInterceptor())
            .setAllowedOrigins(
                "http://localhost:5173",
                "https://wolf.robothanzo.dev"
            )
    }
}
